app/javascript/githubRepos/githubRepos.jsx

```diff
@@ -0,0 +1,59 @@
+import { h, Component } from 'preact';
+import { SingleRepo } from './singleRepo';
+
+export class GithubRepos extends Component {
+  state = {
+    repos: [],
+    erroredOut: false,
+  };
+
+  componentDidMount() {
+    this.getGithubRepos();
+  }
+
+  getGithubRepos = () => {
+    fetch(`/api/github_repos`, {
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      credentials: 'same-origin',
+    })
+      .then(response => response.json())
+      .then(json => {
+        this.setState({ repos: json });
+      })
+      .catch(error => {
+        console.log(error);
+        this.setState({ erroredOut: true });
+      });
+  };
+
+  render() {
+    const { repos, erroredOut } = this.state;
+    const allRepos = repos.map(repo => (
+      <SingleRepo
+        githubIdCode={repo.github_id_code}
+        name={repo.name}
+        fork={repo.fork}
+        selected={repo.selected}
+      />
+    ));
+
+    if (erroredOut) {
+      return (
+        <div className="github-repos github-repos-errored">
+          An error occurred. Please check your browser console and email
+          <a href="mailto:yo@dev.to"> yo@dev.to </a>
+          for more help.
+        </div>
+      );
+    }
+    if (repos.length > 0) {
+      return <div className="github-repos">{allRepos}</div>;
+    }
+    return <div className="github-repos loading-repos" />;
+  }
+}
+
+GithubRepos.displayName = 'GitHub Repos Wrapper';
```

Comment:
Hey @nickytonline, does this look right? My Googling skills didn't seem to find much info on how to do it properly.

Reply:
@Zhao-Andy, actually scratch the `displayName`. My bad. I completely forgot there's a Babel plugin that auto generates it for you. The reason I mentioned it is because with TypeScript and React, you need to do this manually. Sorry for stringing you along ;)

Here's the search stuff I did on localhost and you can see they have display names, e.g. `<SearchForm ... />` where SearchForm is the `displayName`.

![image](https://user-images.githubusercontent.com/833231/47466271-80d4fc00-d7be-11e8-84d2-1244dd9fb54c.png)